sos/policies/runtimes/docker.py

```diff
@@ -18,16 +18,16 @@ class DockerContainerRuntime(ContainerRuntime):
     name = 'docker'
     binary = 'docker'
 
-    def check_is_active(self, sysroot=None):
+    def check_is_active(self):
         # the daemon must be running
-        if (is_executable('docker', sysroot) and
+        if (is_executable('docker', self.policy.sysroot) and
                 (self.policy.init_system.is_running('docker') or
                  self.policy.init_system.is_running('snap.docker.dockerd'))):
             self.active = True
             return True
         return False
 
     def check_can_copy(self):
-        return self.check_is_active(sysroot=self.policy.sysroot)
+        return self.active
```

Comment:
`self.active = False` until `check_is_active` is called (applies both for `ContainerRuntime` and `DockerContainerRuntime`). Can't we get wrong `False` return here just because we haven't called `check_is_active`?

Reply:
In the current flow, no, because a runtime is only loaded if `check_is_active()` returns `True` when the policy is loaded. If this call returns False during policy instantiation, we drop the runtime entirely.

If someone were to separately try to leverage a runtime here, theoretically yes they could end up calling this before `check_is_acitve()`, but that's not something we support.